invenio_records_resources/services/files/schema.py

```diff
@@ -10,12 +10,40 @@
 
 from datetime import timezone
 
-from marshmallow import Schema
+from marshmallow import INCLUDE, Schema
 from marshmallow.fields import UUID, Dict, Number, Str
 from marshmallow_utils.fields import GenMethod, Links, SanitizedUnicode, \
     TZDateTime
 
 
+class InitFileSchema(Schema):
+    """Service (component) schema for file initialization.
+
+    The UI only sends a key and the documentation only refers to a key.
+    The tests though pass other fields.
+
+    Option 1: Only key
+    Pros: We limit what we support, we prevent instances from saving data
+          that we will need to support.
+    Cons: Change a few tests, disable PUT endpoint really
+
+    Option 2: Allow extra fields
+    Pros: Everything stays the same.
+    Cons: The same is loose / not quite consistent.
+
+    Given LTS, going for option 2 so that many changes are not introduced.
+    But ideally option 1 seems better: we can add other fields when we do
+    support third-party data hosting (and perhaps become FileSchema).
```

Comment:
I would put this explanation in the PR, but not commit it to the code. It might stay there for years and not have meaning by then. A shorter version e.g. "Including all values since there is not a defined schema for files metadata" might be enough.

Reply:
I am experimenting here and it would be interesting to keep it. I will be bringing this up in the retrospecitve, but I have found it extremely hard to track down these explanations because they were outside the code (the line history can easily change and honestly it's faster/clearer to have them coupled with the code). Because I (or potential other developers) can't necessarily ask someone during work hours, it also helps with the time difference. 

It can definitely be shorter next time around :sweat_smile: .